cpp/src/parquet/column_reader.cc

```diff
@@ -1369,6 +1369,26 @@ class TypedRecordReader : public TypedColumnReaderImpl<DType>,
     return bytes_for_values;
   }
 
+  const void* ReadDictionary(int32_t* dictionary_length) override {
+    if (this->current_decoder_ == nullptr && !this->HasNextInternal()) {
```

Comment:
Would this trigger initialization? (Called before initialization, since `current_decoder_ == nullptr`)

Reply:
yes.. it would try to initialize it in HasNExtInternal